src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -546,11 +598,11 @@ private void OnAnalyzerException(DiagnosticAnalyzer analyzer, ProjectId projectI
                     exceptionDiagnostic = CompilationWithAnalyzers.GetEffectiveDiagnostics(ImmutableArray.Create(exceptionDiagnostic), compilationOpt).SingleOrDefault();
                 }
 
-                var onAnalyzerException = _owner.GetOnAnalyzerException(projectId);
-                onAnalyzerException(ex, analyzer, exceptionDiagnostic);
+                var onAnalyzerException = service.GetOnAnalyzerException(projectId, logAggregatorOpt);
+                onAnalyzerException?.Invoke(ex, analyzer, exceptionDiagnostic);
```

Comment:
when does GetOnAnalyzerException return null?  can we rename to TryGetOnAnalyzerException?